dpnp/backend/kernels/dpnp_krnl_mathematical.cpp

```diff
@@ -283,7 +283,7 @@ class dpnp_trapz_c_kernel;
 template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
 void dpnp_trapz_c(const void* array1_in, const void* array2_in, void* result1, double dx, size_t array1_size, size_t array2_size)
 {
-    if ((array1_in == nullptr) || (array2_in == nullptr && array1_size > 1))
+    if ((array1_in == nullptr) || (array2_in == nullptr && array2_size > 1))
     {
         return;
     }
```

Comment:
It is better separate this checks.

As for `array2_in`, it can be `None`. You have code for that case.

You may provide something like this:
```cpp
if (array1_in == nullptr || !array1_size)
{
    return;
}

if (array2_in == nullptr && (array2_size > 0))
{
    throw std::runtime_error("DPNP Error: dpnp_trapz_c() failed. Invalid arguments for array2_in");
    // throw std::runtime_error("DPNP Error: dpnp_trapz_c() failed.");  // or just this one
}
```

This would be good solution for backend lib users.
Also be aware `array2_in` from cython level can be non `nullptr`.